patches/server/1065-Create-Limited-set-for-Entity.patch

```diff
@@ -0,0 +1,74 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Intybyte <vaan.testwork@gmail.com>
+Date: Mon, 30 Sep 2024 17:16:36 +0200
+Subject: [PATCH] Create Limited set for Entity
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/PaperLimitedSet.java b/src/main/java/io/papermc/paper/entity/PaperLimitedSet.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..456c8cdb38d29a73a96dc44f7e8e4ce799cbab38
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/PaperLimitedSet.java
+@@ -0,0 +1,40 @@
++package io.papermc.paper.entity;
++
++import org.jetbrains.annotations.NotNull;
++import java.util.Collection;
++import java.util.HashSet;
++
++public class PaperLimitedSet extends HashSet<String> {
++    private final int maxTags;
++
++    public PaperLimitedSet(int maxTags) {
++        this.maxTags = maxTags;
++    }
++
++    @Override
++    public boolean add(String tag) {
++        // Enforce the max tag limit
++        if (this.size() >= maxTags) {
++            return false;
++        }
++        return this.add(tag);
++    }
++
++    @Override
++    public boolean addAll(@NotNull Collection<? extends String> tags) {
++        int currentSize = this.size();
```

Comment:
The `currentSize` field seems redundant? We already have the `size()` method, which is automatically increased anyway